graphs/boruvka.py

```diff
@@ -0,0 +1,210 @@
+"""Borůvka's algorithm.
+
+    Determines the minimum spanning tree(MST) of a graph using the Borůvka's algorithm.
+    Borůvka's algorithm is a greedy algorithm for finding a minimum spanning tree in a
+    graph,or a minimum spanning forest in the case of a graph that is not connected.
+
+    The time complexity of this algorithm is O(ELogV), where E represents the number
+    of edges, while V represents the number of nodes.
+
+    The space complexity of this algorithm is O(V + E), since we have to keep a couple
+    of lists whose sizes are equal to the number of nodes, as well as keep all the
+    edges of a graph inside of the data structure itself.
+
+    Borůvka's algorithm gives us pretty much the same result as other MST Algorithms -
+    they all find the minimum spanning tree, and the time complexity is approximately
+    the same.
+
+    One advantage that Borůvka's algorithm has compared to the alternatives is that it
+    doesn't need to presort the edges or maintain a priority queue in order to find the
+    minimum spanning tree.
+    Even though that doesn't help its complexity, since it still passes the edges logE
+    times, it is a bit more simple to code.
+
+    Details: https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm
+"""
+
+
+class Graph:
+    """Class Graph."""
+
+    def __init__(self, num_of_nodes: int) -> None:
+        """
+        Arguments:
+            num_of_nodes - the number of nodes in the graph
+        Attributes:
+            m_v - the number of nodes in the graph.
+            m_edges - the list of edges.
+            m_component - the dictionary which stores the index of the component which
+            a node belongs to.
+        """
+
+        self.m_v = num_of_nodes
+        self.m_edges = []
+        self.m_component = {}
+
+    def add_edge(self, u_node: int, v_node: int, weight: int) -> None:
+        """Adds an edge in the format [first, second, edge weight] to graph."""
+
+        self.m_edges.append([u_node, v_node, weight])
+
+    def find_component(self, u_node: int) -> int:
+        """Propagates a new component throughout a given component."""
+
+        if self.m_component[u_node] == u_node:
+            return u_node
+        return self.find_component(self.m_component[u_node])
+
+    def set_component(self, u_node: int) -> None:
+        """Finds the component index of a given node"""
+
+        if self.m_component[u_node] == u_node:
```

Comment:
Would be just `self.m_component[u_node] != u_node:` and then we don't need `else`